src/test/resources/com/puppycrawl/tools/checkstyle/checks/coding/unnecessaryparentheses/InputUnnecessaryParenthesesIfStatement2.java

```diff
@@ -49,8 +49,8 @@ public void checkBooleanStatements() {
         if ((b == c) == a
                 && (
                         ( // violation 'Unnecessary parentheses around expression'
-                            (b==c)==(d>=b)==a!=(c==d))
-                    || (b<=c)!=a==(c>=d))) {
+                            (b==c)==d>=b==a!=(c==d))
+                    || b<=c!=a==c>=d)) {
```

Comment:
@Kevin222004 can't we restore these now?

Reply:
yup will update

Reply:
@nrmancuso please let me know if doc update is good will make push for this change

Reply:
Left last item from me below

Reply:
done